### PR TITLE
Improve cli commands

### PR DESCRIPTION
**Pull request by trevorbernard**
Fixes: https://github.com/Granola-Team/mina-indexer/issues/283, https://github.com/Granola-Team/mina-indexer/issues/269

Improving the ergonomics of the CLI interface phase 1.

### CODE REVIEW

File: src/bin/mina-indexer.rs [hunk 1]
```diff
@@ -39,25 +36,14 @@ enum IndexerCommand {
     },
 }
 
-#[derive(Subcommand, Debug)]
-enum ServerCommand {
-    /// Start the mina indexer with a config file
-    Config {
```
Comment (robinbb) [inline src/bin/mina-indexer.rs:45]: Here you are removing functionality that was indeed tested (to some minimal degree). Do we support a config file or no? If no, then surely there is some sort of YAML parsing code that should be deleted. If yes, then how do we specify where the config file is located?

**Review by robinbb — CHANGES_REQUESTED** [review]
The feature of parsing a YAML config file is deleted with this PR. It is not clear to me that we agree that we should not parse a config file, so I suggest splitting this PR into 2: the CLI changes, and then the config file support deletion.

File: src/bin/mina-indexer.rs (hunk 1 above)
Comment (trevorbernard) [reply src/bin/mina-indexer.rs:45]: The functionality is not easily replicated. There was a reason why the additional CLI command was used in favour of command line/config file precedence. For sake of change latency I was going to create another issue to add it back in.

Comment (robinbb) [reply src/bin/mina-indexer.rs:45]: Okay, if you create that issue, refer to it here, and then merge in current 'main', I'll approve and merge.

**Comment by trevorbernard** [comment]
@robinbb Ok, I'll make those changes

**Pull request closed without merging**